src/Core/Checkout/Document/Zugferd/ZugferdBuilder.php

```diff
@@ -58,10 +60,23 @@ public function buildDocument(OrderEntity $order, DocumentConfiguration $config,
             ->withSellerInformation($config)
             ->withDelivery($order->getDeliveries() ?? new OrderDeliveryCollection())
             ->withTaxes($order->getPrice())
-            ->withGeneralOrderData($deliveryDate, $config->getDocumentDate() ?? 'now', $config->getDocumentNumber() ?? '', $order->getCurrency()?->getIsoCode() ?? '');
+            ->withGeneralOrderData($deliveryDate, $config->getDocumentDate() ?? 'now', $config->getDocumentNumber() ?? '', $order->getCurrency()?->getIsoCode() ?? '')
+            ->withBuyerReference($order->getOrderNumber() ?? '');
 
         $this->addLineItems($document, $order->getLineItems());
 
+        $transaction = $order->getTransactions()?->last();
+        if ($transaction !== null) {
+            if ($transaction->getStateMachineState()?->getTechnicalName() === 'paid') {
+                $document->withPaidAmount($order->getAmountTotal());
```

Comment:
Do we support more than one transaction per order? In that case this should check that all of them are marked as "paid" right? And adding payment methods should include all of them if documents support that 🤔